### PR TITLE
Revise kubernetes documentation

### PR DESCRIPTION
**Pull request by puneetse**
This PR:

- Restructures the order of concepts are presented to flow more naturally to a new Kubernetes users

- Consolidates commands to make the document more readable, and influenced from the official Kubernetes documentation wherever possible. 

- Adds instructions for using Dockershim and containerd runtimes.

Related to https://github.com/clearlinux/clear-linux-documentation/issues/894

### CODE REVIEW

**Comment by puneetse** [comment]
@mythi @eadamsintel can you please review?

My intent here is to improve the document given the current state of affairs and make sure it is technically accurate for now. I know there are open discussions on changing the defaults. We can address those in another iteration when they are finalized and commit thee doc improvements now.

File: source/tutorials/kubernetes.rst [hunk 1]
```diff
@@ -3,170 +3,340 @@
...
+For more background information see:
 
-Kata Containers\* kata-runtime adheres to
-:abbr:`OCI (Open Container Initiative*)` guidelines and works seamlessly with
-Kubernetes. `Kata Containers`_ provides strong isolation for untrusted
-workloads or  multi-tenant scenarios. Kata Containers can be
-allocated on a per-pod basis, so you can mix and match both on the same host
-to suit your needs.
+* `What is Kubernetes?`_
+* `What is CRI+O?`_
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: This feels incomplete without containerd and Docker. If you wanted to follow how Kubernetes describes these at https://kubernetes.io/docs/setup/production-environment/container-runtimes/ you would have   Docker/CRI-O/Containerd/Frakti which are CRI runtime replacements.  Kata is an OCI runtime and Docker uses runc which is also an OCI runtime.

File: source/tutorials/kubernetes.rst [hunk 2]
```diff
@@ -3,170 +3,340 @@
...
+     :command:`net.ipv4.ip_forward` parameter
+
+     .. code-block:: bash
+
+        sudo mkdir -p /etc/sysctl.d/
+
+        sudo tee /etc/sysctl.d/99-kubernetes-cri.conf > /dev/null <<EOF
+        net.bridge.bridge-nf-call-iptables  = 1
+        net.ipv4.ip_forward                 = 1
+        net.bridge.bridge-nf-call-ip6tables = 1
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: Is this new guidance?  Previously only `net.ipv4.ip_forward = 1` is required.   Just searching through the web for these settings I found a forum where they also recommended  `net.bridge.bridge-nf-call-arptables = 1`

File: source/tutorials/kubernetes.rst [hunk 3]
```diff
@@ -3,170 +3,340 @@
...
 
-      net.ipv4.ip_forward = 1
 
-   Apply the change:
+#. Disable swap:
 
    .. code-block:: bash
 
-      sudo systemctl restart systemd-sysctl
+      sudo systemctl mask $(sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps) 2>/dev/null
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I would still have a small section header or bolded heading that says "Temporary" and "Permanent"

File: source/tutorials/kubernetes.rst [hunk 4]
```diff
@@ -3,170 +3,340 @@
...
+
+      Kubernetes is designed to work without swap. Performance degradation of other workloads can occur
+      with swap disabled on systems with constrained memory resources.
+
+#. Add the the system's hostname to the :file:`/etc/hosts` file. Kubernetes
+   will read this file to locate the master host.
+
+    .. code-block:: bash
+
+       echo "127.0.0.1 localhost `hostname`" | sudo tee --append /etc/hosts
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: This command if run twice adds the entry twice.

File: source/tutorials/kubernetes.rst [hunk 5]
```diff
@@ -3,170 +3,340 @@
...
 
-         Swap will be enabled at next reboot, causing failures in
-         your cluster if you choose this method.
+In Kubernetes, a `pod
+<https://kubernetes.io/docs/concepts/workloads/pods/pod/>`_ is a group of one
+or more containers and is the smallest deployable unit of computing in a
+Kubernetes cluster. Pods have shared storage/network internally but
+communication between pods requires additional configuration. If you want your
+pods to be able to communicate with each other you must choose and install a
+`pod network add-on`_. Otherwise, this section can be skipped.
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: Is there any case where someone would want a kubernetes cluster but not have a pod network add on?  If not then I would drop the "this section can be skipped" part.

File: source/tutorials/kubernetes.rst [hunk 6]
```diff
@@ -3,170 +3,340 @@
...
+This is important to decide early because some pod network add-ons require
+configuration during cluster initialization. Check whether or not your add-on
+requires special flags when you initialize the master control plane.
 
-   b) Permanently:
+If your chosen network add-on requires appending to the :command:`kubeadm
+init` command, make note of it before continuing. 
 
-      Mask the swap partition:
+For example, if you choose the *flannel* pod network add-on, then in later
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: Id also give an example where you don't have to change the settings below.

File: source/tutorials/kubernetes.rst [hunk 7]
```diff
@@ -3,170 +3,340 @@
...
-#. Switch to root to modify the `hosts` file:
+Container runtimes are the underlying fabric that pod workloads execute inside
+of. Different container runtimes offer different balances between features,
+performance, and security. 
+
+Kubernetes allows integration various container runtimes via a container
+runtime interface (CRI). |CL| supports Kubernetes with the various runtimes
+below:
+
+* `Docker (Dockershim)`_ with or without `Kata Containers`_.
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: This is kind of a weird way to state with or without Kata containers for Docker.  Maybe for each of these say. It is just a suggestion.

Docker with fixed OCI runtime by Node
containerd with selectable OCI runtime by pod (runc or Kata Containers)
CRI-O with  selectable OCI runtime by pod (runc or Kata containers)

File: source/tutorials/kubernetes.rst [hunk 8]
```diff
@@ -3,170 +3,340 @@
...
+initialize the master cluster with :command:`kubeadm init`.
+
+
+Docker (Dockershim)
+-------------------
+
+The default built-in runtime provided by Kubernetes is using the system Docker
+via Dockershim and as a result is one of the simplest to use. 
+
+One limitation of using Dockershim is that all pods on the Kubernetes node
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I think this should be combined with the previous sentence to make a paragraph.

File: source/tutorials/kubernetes.rst [hunk 9]
```diff
@@ -3,170 +3,340 @@
...
+   .. important:: 
+
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. In this example, *flannel* is
+      being used for the pod networking. See `Choose a pod network add-on`_
+      for more information.
+
+   .. code:: bash
+
+      sudo kubeadm init \
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I think showing two examples both with and without would be more helpful.

File: source/tutorials/kubernetes.rst [hunk 10]
```diff
@@ -3,170 +3,340 @@
...
+      sudo systemctl mask docker-set-default-runtime.service
+
+#. Once the cluster initialization is complete, continue reading about how to
+   `Use your cluster`_.
+
+containerd
+----------
+
+`containerd <https://containerd.io/>`_ is the runtime  that the Docker engine
+is built ontop of. Kubernetes can use containerd directly instead of going
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: "on top"

File: source/tutorials/kubernetes.rst [hunk 11]
```diff
@@ -3,170 +3,340 @@
...
+      sudo rm /etc/systemd/system/docker.service.d/50-runtime.conf
+      sudo systemctl mask docker-set-default-runtime.service
+
+#. Once the cluster initialization is complete, continue reading about how to
+   `Use your cluster`_.
+
+containerd
+----------
+
+`containerd <https://containerd.io/>`_ is the runtime  that the Docker engine
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: It is weird to start a sentence with a lowercase word. I'd reword so you don't start that way.

File: source/tutorials/kubernetes.rst (hunk 1 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: I agree, it does seem out of place. I originally pulled out sections based on them being too much background information embedded in sections.  CRI+O was one of these.

Looking at it now, I actually think it would be better to simplify this to just 2: What is Kubernetes? and What is Kata?. The rest of the container runtime descriptions are small enough to embed, I think.

File: source/tutorials/kubernetes.rst (hunk 2 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Yes, according to https://kubernetes.io/docs/setup/production-environment/container-runtimes/ and the cloud-native-scripts.

File: source/tutorials/kubernetes.rst (hunk 3 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: I'm opinionated that the temporary step was noise in the documentation and not useful for most people who are going to be using a system for a k8s node. I used the temporary steps and then at next reboot it failed citing swap was enabled so it was not useful to me.

If you think there is going to be a population that wants to do this temporarily, then I rather keep it as a single step + note to not run the mask command.

File: source/tutorials/kubernetes.rst [hunk 12]
```diff
@@ -3,170 +3,340 @@
...
 
-Configure and run CRI-O + kata-runtime
-======================================
+   .. important:: 
 
-#.  Enable the CRI-O service:
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. In this example, *flannel* is
+      being used for the pod networking. See `Choose a pod network add-on`_
+      for more information.
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I think having two examples would be helpful here as well.

File: source/tutorials/kubernetes.rst (hunk 4 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: True. We can make it replace the whole file, but there are many documents in the tree like this that kind of assume if you've done it once, you don't blindly repeat it, for sake of simplicity.

File: source/tutorials/kubernetes.rst [hunk 13]
```diff
@@ -3,170 +3,340 @@
...
-       sudo systemctl enable crio.service
+      sudo kubeadm init \
+      --cri-socket=/run/containerd/containerd.sock \ #required for containerd
+      --pod-network-cidr 10.244.0.0/16   #required for flannel
 
-#.  Enter these commands to restart the system services so that CRI-O will be restarted, and will automatically start with every reboot.
 
-    .. code-block:: bash
+#. (Optional) By default, containerd will use the default runc as the default
+   runtime. The with or without `Kata Containers`_. runtime can be set
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: Sentence fragment

File: source/tutorials/kubernetes.rst [hunk 14]
```diff
@@ -3,170 +3,340 @@
...
+   `Use your cluster`_.
 
 
+CRI+O
+-----
 
-#.  Initialize the master control plane with the command below and follow the displayed instructions to set up `kubectl`.  The :command:`init` command initializes a Kubernetes `control-plane node`_. If you will be adding a pod network add-on, you will need to add configuration to the :command:`init` command line. see the `Install pod network add-on`_ section for details.
+`CRI+O <https://cri-o.io/>`_ is a popular runtime choice because it is
+lightweight and flexible. CRI+O can use OCI compliant runtimes, such as runc
+and kata, and allows specifying runtimes per-Kubernetes service.
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I think saying "specifying runtimes on a per pod basis" might be a better way of saying it.  Do all kubernetes services fit neatly into pods? I'm not sure.

File: source/tutorials/kubernetes.rst [hunk 15]
```diff
@@ -3,170 +3,340 @@
...
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. In this example, *flannel* is
+      being used for the pod networking. See `Choose a pod network add-on`_
+      for more information.
+
+   .. code-block:: bash
+
+      sudo kubeadm init \
+      --cri-socket=/run/crio/crio.sock \ #required for CRI+O
+      --pod-network-cidr 10.244.0.0/16   #required for flannel
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: This is confusing. It looks like it is all one command.  I'd make it two mini sections with different commands and follow the same format you would use for docker and containerd.

File: source/tutorials/kubernetes.rst (hunk 7 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: I tried wording this a lot of different ways and am open to changing it. 

From a new reader perspective, it has too many technical details which don't really help in context of introducing the runtimes.  The details you mentioned are further in the sections after the different options are introduced.

File: source/tutorials/kubernetes.rst [hunk 16]
```diff
@@ -180,71 +350,143 @@ Configure and run CRI-O + kata-runtime
...
+   network add-on`_), with CRI-O + Kata Containers as the runtime, edit the
+   :file:`/etc/crio/crio.conf` file to include:
 
-Install pod network add-on
-**************************
+   .. code-block:: console
 
-You must choose and install a `pod network add-on`_ to allow your pods to communicate. Check whether or not your add-on requires special flags when you initialize the master control plane.
+      [crio.runtime]
+      manage_network_ns_lifecycle = true
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: This has been the default setting when you enable the crio system service.  I would specifically point out that flannel is the only exception if that is true. I am not sure if it is.

File: source/tutorials/kubernetes.rst [hunk 17]
```diff
@@ -3,170 +3,340 @@
...
-#.  Initialize the master control plane with the command below and follow the displayed instructions to set up `kubectl`.  The :command:`init` command initializes a Kubernetes `control-plane node`_. If you will be adding a pod network add-on, you will need to add configuration to the :command:`init` command line. see the `Install pod network add-on`_ section for details.
+`CRI+O <https://cri-o.io/>`_ is a popular runtime choice because it is
+lightweight and flexible. CRI+O can use OCI compliant runtimes, such as runc
+and kata, and allows specifying runtimes per-Kubernetes service.
 
-    .. code-block:: bash
+This section describes how to configure and run Kubernetes with CRI-O and
+kata-runtime.
+
+#. Start the CRI-O service and enable it to run at boot automatically:
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I would mention that when you run that enable command it creates  a default /etc/crio/crio.conf file that will work on Clear Linux.

File: source/tutorials/kubernetes.rst [hunk 18]
```diff
@@ -3,170 +3,340 @@
...
+      sudo kubeadm init \
+      --cri-socket=/run/containerd/containerd.sock \ #required for containerd
+      --pod-network-cidr 10.244.0.0/16   #required for flannel
 
-#.  Enter these commands to restart the system services so that CRI-O will be restarted, and will automatically start with every reboot.
 
-    .. code-block:: bash
+#. (Optional) By default, containerd will use the default runc as the default
+   runtime. The with or without `Kata Containers`_. runtime can be set
+   per-pod. See the Kata Containers documentation on `creating an untrusted
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I just confirmed the Kata documentation is old and out of date.  The containerd documentation should mention runtime class as well and describe how to do it that way.  I think you could still link to it but remove mentions of trusted/untrusted so that when that doc is updated these instructions still work. I *think* that still works by using it as trusted/untrusted but it is not the official way to do things anymore.

File: source/tutorials/kubernetes.rst (hunk 9 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Do you mean one example without flannel and one with? If so, that is one of the things I wanted to simplify. Examples trying to cover all the different cases is hard to follow.

File: source/tutorials/kubernetes.rst [hunk 19]
```diff
@@ -180,71 +350,143 @@ Configure and run CRI-O + kata-runtime
...
+cluster with additional nodes and start deploying containerized applications.
+For further information on using Kubernetes, see `Related topics`_. 
 
-Related topics
-**************
+.. note:: 
 
-Read the Kubernetes documentation to learn more about:
+   By default, the master node does not run any pods for security reasons. To
+   setup a single-node cluster and  allow the master node to also run pods,
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: Id change to  "To setup a single-node cluster and  allow the master node to also run pods, the master node will need its taints removed. See the ..."

File: source/tutorials/kubernetes.rst (hunk 10 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Thanks. Since this is editorial, I'll prefer TCS to do this in a follow up PR.

File: source/tutorials/kubernetes.rst (hunk 11 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Thanks. Since this is editorial, I'll prefer TCS to do this in a follow up PR.

File: source/tutorials/kubernetes.rst (hunk 13 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Whoops. Thanks. Will fix in this PR.

File: source/tutorials/kubernetes.rst [hunk 20]
```diff
@@ -180,71 +350,143 @@ Configure and run CRI-O + kata-runtime
...
+supports runc and Kata Containers as the container runtimes but any
+OCI-conformant runtime can be plugged in principle.
 
-* `Deploying an application to your cluster`_
+
+What is Kata Containers?
+========================
+
+`Kata Containers`_ is a secure container runtime leveraging lightweight
+virtual machines that feel and perform like containers, but provide stronger
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: Kata Containers is an alternative OCI compatible runtime that secures container workloads in a lightweight virtual machine.  It provides stronger workloads isolation using hardware virtualization technology as a second layer of defense for untrusted workloads or multi-tenant scenarios.

File: source/tutorials/kubernetes.rst [hunk 21]
```diff
@@ -180,71 +350,143 @@ Configure and run CRI-O + kata-runtime
...
+workload isolation using hardware virtualization technology as a second layer
+of defense for untrusted workloads or multi-tenant scenarios.
+
+Kata Containers can be allocated on a per-pod basis, so you can mix and match
+both on the same host to suit your needs.
+
+The Kata Containers\* (kata-runtime) adheres to :abbr:`OCI (Open Container
+Initiative*)` guidelines and works seamlessly with Kubernetes through CRI+O.
+
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: Where is containerd?

File: source/tutorials/kubernetes.rst [hunk 22]
```diff
@@ -180,71 +350,143 @@ Configure and run CRI-O + kata-runtime
...
 
-If you customize any of the default package configuration files, you **must** store the customized files in the :file:`/etc/` directory. If you edit any of the distribution-provided default files, your changes will be lost in the next system update as the default files will be overwritten with the updated files.
+If you customize any of the default package configuration files, you **must**
+store the customized files in the :file:`/etc/` directory. If you edit any of
+the distribution-provided default files, your changes will be lost in the next
+system update as the default files will be overwritten with the updated files.
 
-For example, to customize CRI-O configuration in your system, run the following commands:
+For example, to customize CRI-O configuration in your system, run the
+following commands:
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: This code block explanation seems redundant.  When you enable the crio service it automatically creates /etc/crio/crio.conf.  You don't have to copy it.

File: source/tutorials/kubernetes.rst (hunk 12 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Do you mean one example without flannel and one with? If so, that is one of the things I wanted to simplify. Examples trying to cover all the different cases is hard to follow.

File: source/tutorials/kubernetes.rst (hunk 15 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: It is one big command with in-line comment. Multiple code-blocks are even messier.

File: source/tutorials/kubernetes.rst (hunk 14 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Sounds good. I'll change it.

File: source/tutorials/kubernetes.rst (hunk 15 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: I'll also add that it renders nicely with the CSS of the website.

File: source/tutorials/kubernetes.rst (hunk 16 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Yeah, I was waffling on this too and how to fit it in. The exception cases are what makes it hard to word but will give it another shot.

File: source/tutorials/kubernetes.rst (hunk 17 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Good idea. I'll add that.

File: source/tutorials/kubernetes.rst [hunk 23]
```diff
@@ -255,15 +497,23 @@ For example, to customize CRI-O configuration in your system, run the following
 Learn more about :ref:`stateless` in |CL|.
 
 Proxy configuration (optional)
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: It seems like a real burden to try to explain proxies in here. I don't think this will cover every scenario.  I know in our lab setups with multiple nodes you have to add each node's ip to the `hosts` file to deal with proxies.

File: source/tutorials/kubernetes.rst [hunk 24]
```diff
@@ -255,15 +497,23 @@ For example, to customize CRI-O configuration in your system, run the following
...
+enough.)
 
-If you have already set your proxy environment variables, run the following commands as a shell script to configure all of these services in one step:
+If you have already set your proxy environment variables, run the following
+commands as a shell script to configure all of these services in one step:
 
 .. code-block:: bash
 
-   services=('crio')
+   services=(kubelet crio containerd)
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: What about Docker proxies?

File: source/tutorials/kubernetes.rst (hunk 18 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Thanks. Can you post the link showing how to set the RuntimeClass?

File: source/tutorials/kubernetes.rst [hunk 25]
```diff
@@ -275,77 +525,91 @@ If you have already set your proxy environment variables, run the following comm
...
-Troubleshooting
-***************
+
+Below are some common symptoms and solutions to problems:
 
 * <HOSTNAME> not found in <IP> message.
 
-  Your DNS server may not be appropriately configured. Try adding an entry to the :file:`/etc/hosts` file with your host's IP and Name.
+   Your DNS server may not be appropriately configured. Try adding an entry
+   to the :file:`/etc/hosts` file with your host's IP and Name.
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I am pretty sure this needs to go in the no_proxy variable.

File: source/tutorials/kubernetes.rst (hunk 21 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: What do you mean? These section is about Kata. Can you elaborate on what is missing?

File: source/tutorials/kubernetes.rst (hunk 22 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: I agree, this was existing content.

File: source/tutorials/kubernetes.rst (hunk 23 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: I agree, this was existing content so I just did my best to format it nicely.

File: source/tutorials/kubernetes.rst [hunk 26]
```diff
@@ -275,77 +525,91 @@ If you have already set your proxy environment variables, run the following comm
...
+   to set the proxy variables permanently, and how to make them available for
+   all the types of access that you will use, such as remote SSH access.
 
-  If the result of the above commands is blank, you may need to add a ``profile`` to the :file:`/etc` directory. To do so, follow these steps.
+   If the result of the above commands is blank, you may need to add a
+   ``profile`` to the :file:`/etc` directory to set environment variables
+   persistently. To do so, follow these steps.
 
-  #. Create a `profile` in :file:`/etc`
+   #. Create a `profile` in :file:`/etc`
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: These set of instructions are confusing.  I think you meant to say `cd /etc` first before running `sudo touch profile`  The `cloud-native-setup` [script](https://github.com/clearlinux/cloud-native-setup/blob/00df885b45322a1d3b56562fef908f1c960b4690/clr-k8s-examples/setup_system.sh#L153) checks `/etc/profile.d` directory for proxy environment variables. That is probably a better directory to put this information in.

File: source/tutorials/kubernetes.rst (hunk 24 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Thanks. Added.

File: source/tutorials/kubernetes.rst (hunk 26 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: This was existing content that I didn't check (the whole troubleshooting section). I'll update it in anycase but we can decide how to revise or pull out this section as a whole later.

File: source/tutorials/kubernetes.rst (hunk 2 above)
Comment (eadamsintel) [reply source/tutorials/kubernetes.rst]: OK then.  What about arptables?  I saw that in some forum threads in addition to the above.  I think the above are when running kubernetes in a virtual machine already but am not 100% sure.

File: source/tutorials/kubernetes.rst (hunk 3 above)
Comment (eadamsintel) [reply source/tutorials/kubernetes.rst]: I think then that putting the `sudo swapoff -a` above the `mask` step with a small header that says `Disable Temporarily` and then right below that have `Disable Permanently` as a small header above the `mask` step.

File: source/tutorials/kubernetes.rst (hunk 9 above)
Comment (eadamsintel) [reply source/tutorials/kubernetes.rst]: I meant show one example where you init and don't have to add any extra parameters and then another example where you do add the extra parameters.  When you reference that pod network add-on link you can see how the extra parameters fit into the overall init command.   I'm not sure how you could fit this into the doc but seeing 5 or 6 different init commands at one time showing how the different CRI runtimes are started and how different network plugins are enabled are always super helpful for me.  It allows my brain to easily see what is the Kubernetes portion of the init, what is the crio/docker/containerd portion of the init, and what are the networking addin portions of the init and how they can vary.

File: source/tutorials/kubernetes.rst (hunk 12 above)
Comment (eadamsintel) [reply source/tutorials/kubernetes.rst]: Yes I meant two example. Preferably the two most popular ones that are different would help.

File: source/tutorials/kubernetes.rst (hunk 18 above)
Comment (eadamsintel) [reply source/tutorials/kubernetes.rst]: Nope. It would be similar to how crio does it but I have not done that yet.  You would have to look at how crio does it and refer to the kubernetes doc on runtime class for the difference.  I can do this in the Kata doc some time after KubeCon but it will take me time to also figure it out.  For this doc you can just refer to the Kata repo as that file would stay the same and just be updated.

File: source/tutorials/kubernetes.rst (hunk 15 above)
Comment (eadamsintel) [reply source/tutorials/kubernetes.rst]: OK It was how I was seeing it.   This is why for me personally I like seeing multiple init commands for different features. I know it makes the doc a bit harder to read in some cases but it helps me understand more quickly how these commands work.

File: source/tutorials/kubernetes.rst (hunk 21 above)
Comment (eadamsintel) [reply source/tutorials/kubernetes.rst]: I meant to put this comment right above where you comment on what cri-o is. You mention cri-o but don't mention containerd.

File: source/tutorials/kubernetes.rst (hunk 1 above)
Comment (eadamsintel) [reply source/tutorials/kubernetes.rst]: Makes sense.  I think either you need a comprehensive list or a very focused list.  Putting some CRI runtimes and not others is confusing.  I have a similar comment at the end of the doc where these link to.

**Comment by puneetse** [comment]
I pushed an update addressing a lot of the feedback. Most notably culling the troubleshooting section.

**Comment by mvincerx** [comment]
@puneetse. Do you have a release plan for this doc? These revisions are evolving nicely. 
I want to check in on your release plan because you mentioned last Monday that you didn't plan to submit/release anything before the conference. Please advise.

**Comment by puneetse** [comment]
Hi @mvincerx! I am primary focused on technical accuracy in the comments of this PR because I think the revisions improving readability of this document are substantial and would be nice to have soon. So I'm listening to and absorbing all feedback but plan to very selectively apply them.

The main thing left I'd like to address is this comment by @eadamsintel https://github.com/clearlinux/clear-linux-documentation/pull/917#discussion_r346040452 . Then if @eadamsintel and @mythi agree it is acceptable, I'd like to merge with any non-techincal/preferential changes following up later.

**Comment by mvincerx** [comment]
> The main thing left I'd like to address is this comment by @eadamsintel [#917 (comment)](https://github.com/clearlinux/clear-linux-documentation/pull/917#discussion_r346040452) . Then if @eadamsintel and @mythi agree it is acceptable, I'd like to merge with any non-techincal/preferential changes following up later.

Hey @puneetse, I understand your approach. That's fine. Just checking in to assure that this is merged by the time of conference--if that's your goal.  Also, you probably already know about this description of [Runtime class for containerd](https://kubernetes.io/docs/concepts/containers/runtime-class/#containerd-https-containerd-io), but I include here anyways. I realize that it will need more config than what's shown. I'll keep my eyes posted on comments from @eadamsintel and @mythi.

**Comment by puneetse** [comment]
Unfortunately, I'm cancelling this PR for now until more testing has been done.
